sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -343,6 +367,25 @@ class JacksonParser(
       record: T,
       createParser: (JsonFactory, T) => JsonParser,
       recordLiteral: T => UTF8String): Seq[InternalRow] = {
+    parseWithArrayOfPrimitiveSupport(record, createParser, recordLiteral) match {
+      case rows: Seq[InternalRow] => rows
+      case _: Seq[_] => throw BadRecordException(() => recordLiteral(record), () => None,
+        new RuntimeException("Conversion of array of primitive data is not yet supported here."))
```

Comment:
This exception looks a bit weird. How about `` `parse` is only used to parse the JSON input to the set of `InternalRow`s. Use `parseWithArrayOfPrimitiveSupport` when paring array of primitive data is needed``?

Reply:
What about a user seeing this exception? With the current description (which I am very open to improve), he/she is aware that he/she is trying to do something which is not allowed (at least at the moment), ie. we might hit this exception when using `sqlContext.read.json(...)` on arrays of primitives. Your suggested description would be a bit weird to a user: he/she might feel he/she is doing something wrong to achieve something which can be done, but of course he/she knows nothing about these functions so he/she would be lost IMHO.

Reply:
What I thought is `JacksonParser` is internally used in Spark SQL.  It is hard to think an end user will directly use `parse` and see this exception.

Actually `parse` is supposed to return `InternalRow`s. The case we get others is only because the given schema to `JacksonParser` is wrong. So I expect this exception is only seen at SQL development internally.

Btw, I've no strong option at this point. If you think it is ok. I'm fine with it.

Reply:
Yes, it is internally used, but it it throws an `Exception` a user might see it. I think that if this is not clear for internal usage we can add comments, but the text of the exception should be meaningful to the end user IMHO. If you have any suggestion about how to improve this message keeping it meaningful to a user, I am happy to change it. Thanks.

Reply:
To clarify it, I think the only way we throw this exception is passing an `ArrayType` into `JacksonParser` constructor and call `parse` instead of `parseWithArrayOfPrimitiveSupport`. Because `JacksonParser` is internally used, I assume this usage is intentional and the developer will get the exception right away.

So I didn't think this exception will possibly be seen by end user, unless we ship such broken codes to users in Spark releases.

Reply:
Anyway, we can keep it as it is. I didn't feel strongly to change it.

Reply:
yes, you are right since we have only two constructors which enforce this patters. Then I will edit the exception message according to your suggestion, thanks.